### PR TITLE
React.Component translated

### PR DESCRIPTION
**Pull request by hosseinAMD**
OK I'm sorry @sorousht  for some conflict.
because I've made a mistake and forgot to change branch when I'm creating my pull request and I've requested to merge my `master ` branch!

So this is the first one with correct branch.

### CODE REVIEW

File: content/docs/reference-react-component.md [hunk 1]
```diff
@@ -476,73 +476,73 @@ React doesn't call `UNSAFE_componentWillReceiveProps()` with initial props durin
...
 
-`setState()` does not always immediately update the component. It may batch or defer the update until later. This makes reading `this.state` right after calling `setState()` a potential pitfall. Instead, use `componentDidUpdate` or a `setState` callback (`setState(updater, callback)`), either of which are guaranteed to fire after the update has been applied. If you need to set the state based on the previous state, read about the `updater` argument below.
+`setState()` همیشه در لحظه کامپوننت را بروز نمی‌کند. ممکن است تغییر را بسته‌بندی کرده و یا به تاخیر انداخته باشد. این موضوع خواندن `this.state` درست پس از صدا زدن `setState()` را به یک تله تبدیل کرده‌است. به جای اینکار، از `componentDidUpdate` یا یک `setState` callback (`setState(updater, callback)`) استفاده کنید، که اجرا شدن هر دو پس از اینکه بروزرسانی اعمال شده است، تضمین شده است. اگر نیاز به ست کردن state بر اساس state قبلی دارید، در مورد آرگومان `updater` در زیر بخوانید.
```
Comment (sorousht) [inline content/docs/reference-react-component.md:511]:
```suggestion
`setState()` همیشه در لحظه کامپوننت را بروز نمی‌کند. ممکن است تغییر را بسته‌بندی کرده و یا به تاخیر انداخته باشد. این موضوع خواندن `this.state` درست پس از صدا زدن `setState()` را به یک تله تبدیل کرده‌است. به جای اینکار، از `componentDidUpdate` یا یک `setState` callback (`setState(updater, callback)`) استفاده کنید، که اجرا شدن هر دو پس از اینکه به‌روز رسانی اعمال شده است، تضمین شده است. اگر نیاز به ست کردن state بر اساس state قبلی دارید، در مورد آرگومان `updater` در زیر بخوانید.
```

File: content/docs/reference-react-component.md [hunk 2]
```diff
@@ -553,19 +553,19 @@ Object.assign(
...
 
-For more detail, see:
+برای اطلاعات بیشتر موارد زیر را ببینید:
 
-* [State and Lifecycle guide](/docs/state-and-lifecycle.html)
-* [In depth: When and why are `setState()` calls batched?](https://stackoverflow.com/a/48610973/458193)
-* [In depth: Why isn't `this.state` updated immediately?](https://github.com/facebook/react/issues/11527#issuecomment-360199710)
+* [راهنمای State و چرخه‌ی حیات](/docs/state-and-lifecycle.html)
+* [در اعماق: کی و چرا صدا زدن‌های `setState()` دسته‌بندی می‌شوند؟ ](https://stackoverflow.com/a/48610973/458193)
+* [در اعماق: چرا `this.state` فورا بروزرسانی نمی‌شود؟ ](https://github.com/facebook/react/issues/11527#issuecomment-360199710)
```
Comment (sorousht) [inline content/docs/reference-react-component.md:568]:
```suggestion
* [در اعماق: چرا `this.state` فورا به‌روز رسانی نمی‌شود؟ ](https://github.com/facebook/react/issues/11527#issuecomment-360199710)
```

File: content/docs/reference-react-component.md [hunk 3]
```diff
@@ -274,22 +274,22 @@ Currently, if `shouldComponentUpdate()` returns `false`, then [`UNSAFE_component
...
 
-This method exists for [rare use cases](/blog/2018/06/07/you-probably-dont-need-derived-state.html#when-to-use-derived-state) where the state depends on changes in props over time. For example, it might be handy for implementing a `<Transition>` component that compares its previous and next children to decide which of them to animate in and out.
+این متد برای [موارد استفاده نادر](/blog/2018/06/07/you-probably-dont-need-derived-state.html#when-to-use-derived-state) که state وابسته به تغییرات props در طول زمان است وجود دارد. برای مثال، ممکن است برای پیاده‌سازی یک کامپوننت `<Transition>` که فرزند قبلی و بعدی خود را برای تصمیم‌گیری اینکه برای کدام‌یک انیمیشن ورود و برای کدام‌یک انیمیشن خروج را اعمال کند مقایسه می‌کند، کاربردی باشد.
 
-Deriving state leads to verbose code and makes your components difficult to think about.  
-[Make sure you're familiar with simpler alternatives:](/blog/2018/06/07/you-probably-dont-need-derived-state.html)
+اشتقاق state باعث بلندی کد می‌شود و فکر کردن در مورد کامپوننت شما را سخت می‌کند.
```
Comment (hosseinAMD) [inline content/docs/reference-react-component.md:281]:
```suggestion
اشتقاق state باعث بلندی کد می‌شود و فکر کردن در مورد کامپوننت شما را سخت می‌کند.
```

[the remark this answers is not in the document]
Comment (hosseinAMD) [reply content/docs/reference-react-component.md:281]: Your suggestion includes a typo so I've committed correct edit.

File: content/docs/reference-react-component.md [hunk 4]
```diff
@@ -274,22 +274,22 @@ Currently, if `shouldComponentUpdate()` returns `false`, then [`UNSAFE_component
...
+[مطمئن شوید با جایگزین‌های ساده‌تر آشنا هستید:](/blog/2018/06/07/you-probably-dont-need-derived-state.html)
 
-* If you need to **perform a side effect** (for example, data fetching or an animation) in response to a change in props, use [`componentDidUpdate`](#componentdidupdate) lifecycle instead.
+* اگر نیاز به **انجام یک تاثیر بیرونی** (برای مثال fetch کردن داده یا اعمال انیمیشن) در پاسخ یک تغییر در props دارید، از متد چرخه‌ی حیات [`componentDidUpdate`](#componentdidupdate) به عنوان جایگزین استفاده کنید.
```
Comment (hosseinAMD) [inline content/docs/reference-react-component.md:284]:
```suggestion
* اگر نیاز به **انجام یک تاثیر جانبی (side effect)** (برای مثال fetch کردن داده یا اعمال انیمیشن) در پاسخ یک تغییر در props دارید، از متد چرخه‌ی حیات [`componentDidUpdate`](#componentdidupdate) به عنوان جایگزین استفاده کنید.
```

[the remark this answers is not in the document]
Comment (hosseinAMD) [reply content/docs/reference-react-component.md:284]: Your suggestion includes a typo so I've committed correct edit.

File: content/docs/reference-react-component.md [hunk 5]
```diff
@@ -299,41 +299,41 @@ Note that this method is fired on *every* render, regardless of the cause. This
...
+در مثال بالا، خواندن ویژگی `scrollHeight` در `getSnapshotBeforeUpdate` مهم است چرا که ممکن است تاخیرهایی بین فاز "رندر شدن" چرخه‌ی حیات (مانند `render`) و فاز "commit" چرخه‌ی حیات (مانند `getSnapshotBeforeUpdate` و `componentDidUpdate`) وجود داشته باشد.
 
 * * *
 
 ### Error boundaries {#error-boundaries}
 
-[Error boundaries](/docs/error-boundaries.html) are React components that catch JavaScript errors anywhere in their child component tree, log those errors, and display a fallback UI instead of the component tree that crashed. Error boundaries catch errors during rendering, in lifecycle methods, and in constructors of the whole tree below them.
+[Error boundaries](/docs/error-boundaries.html) کامپوننت‌های ری‌اکت هستند که خطاهای جاوااسکریپت را هر جایی در درخت کامپوننت‌های فرزندشان می‌گیرند، آن ارورها را در کنسول در قالب لاگ نشان می‌دهند و یک UI به جای درخت کامپوننتی که کرش کرده است نمایش می‌دهد. Error boundaries خطاها را هنگام رندر شدن، در متدهای چرخه‌ی حیات و در سازنده‌های تمام درخت زیر آن‌ها می‌گیرند.
```
Comment (hosseinAMD) [inline content/docs/reference-react-component.md:318]:
```suggestion
[مرزهای خطا](/docs/error-boundaries.html) کامپوننت‌های ری‌اکت هستند که خطاهای جاوااسکریپت را هر جایی در درخت کامپوننت‌های فرزندشان می‌گیرند، آن خطا‌ها را در کنسول در قالب لاگ نشان می‌دهند و یک UI به جای درخت کامپوننتی که با خطا مواجه شده‌است، نمایش می‌دهد. مرزهای خطا هنگام رندر شدن، در متدهای چرخه‌ی حیات و در سازنده‌های تمام درخت کامپوننت‌های زیر خود، خطا ها را می‌گیرند.
```

[the remark this answers is not in the document]
Comment (hosseinAMD) [reply content/docs/reference-react-component.md:318]: I guess you wrapped `[کامپوننت‌های]` accidentally! So I've committed your changes without brackets.

File: content/docs/reference-react-component.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sorousht) [reply content/docs/reference-react-component.md]: Good catch!

File: content/docs/reference-react-component.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (sorousht) [reply content/docs/reference-react-component.md]: Same here :)

File: content/docs/reference-react-component.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (sorousht) [reply content/docs/reference-react-component.md]: :+1:

**Pull request merged**